pkg/engine/lifecycletest/pulumi_test.go

```diff
@@ -5393,6 +5393,110 @@ func TestResourceNames(t *testing.T) {
 func TestSourcePositions(t *testing.T) {
 	t.Parallel()
 
+	loaders := []*deploytest.ProviderLoader{
+		deploytest.NewProviderLoader("pkgA", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{
+				CreateF: func(urn resource.URN, inputs resource.PropertyMap, timeout float64,
+					preview bool,
+				) (resource.ID, resource.PropertyMap, resource.Status, error) {
+					return "created-id", inputs, resource.StatusOK, nil
+				},
+				ReadF: func(urn resource.URN, id resource.ID,
+					inputs, state resource.PropertyMap,
+				) (plugin.ReadResult, resource.Status, error) {
+					return plugin.ReadResult{Inputs: inputs, Outputs: state}, resource.StatusOK, nil
+				},
+				ConstructF: func(monitor *deploytest.ResourceMonitor,
+					typ, name string, parent resource.URN, inputs resource.PropertyMap,
+					options plugin.ConstructOptions,
+				) (plugin.ConstructResult, error) {
+					// The position of the root component should be replaced with the position of the corresponding call
+					// to RegisterResource in the user program (see compPos below).
+					urn, _, _, err := monitor.RegisterResource(tokens.Type(typ), name, false, deploytest.ResourceOptions{
+						Parent:         parent,
+						SourcePosition: "/component/source/positions#7,8",
+					})
+					require.NoError(t, err)
+
+					// The position of this resource should be elided, as it is not part of the user program.
+					_, _, _, err = monitor.RegisterResource("pkgA:m:typA", "resD", true, deploytest.ResourceOptions{
+						Parent:         urn,
+						SourcePosition: "/component/source/positions#9,10",
```

Comment:
As above. Doesn't matter for this test as is, but if we start doing remapping we want to make sure the source data looks as we expect.